src/fsharp/ConstraintSolver.fs

```diff
@@ -1548,9 +1575,11 @@ and AddConstraint (csenv:ConstraintSolverEnv) ndeep m2 trace tp newConstraint  =
 
                   let rty1 = GetFSharpViewOfReturnType g rty1
                   let rty2 = GetFSharpViewOfReturnType g rty2
-                  Iterate2D (SolveTypeEqualsTypeKeepAbbrevs csenv ndeep m2 trace) argtys1 argtys2 ++ (fun () -> 
-                      SolveTypeEqualsTypeKeepAbbrevs csenv ndeep m2 trace rty1 rty2 ++ (fun () -> 
-                         CompleteD))
+                  trackErrors{
```

Comment:
Like all the other cases put the `trackErrors { .. }` at the function definition for `AddConstraint `